test/face/scan/test_scan.py

```diff
@@ -0,0 +1,476 @@
+        ) for name, team in __artilist__.items()
+    ]
+)
+def test_scan_arti_drop(scantest, name, rare, part, part_type) -> None:
+    """
+    Test the configuration of artifact on the user interface
+
+    :return:
+    """
+
+    """
+    Set the user interface elements as intended
+    """
+    conf = dict()
+
+    conf["dist"] = part_type
+    scantest.arti_dist.setCurrentText(conf["dist"])
+    conf["name"] = name
+    scantest.arti_type.setCurrentText(conf["name"])
+    conf["rare"] = choice([item for item in rare])
+    scantest.arti_rare.setCurrentText(conf["rare"].value.name)
+    conf["levl"] = choice([item for item in ArtiLevl if conf["rare"] in item.value.rare])
+    scantest.arti_levl.setCurrentText(conf["levl"].value.name)
+    conf["stat"] = choice([item for item in getattr(base, __dist__[conf["dist"]]["list"].__name__) if item.value != STAT.none])
+    scantest.arti_name_main.setCurrentText(conf["stat"].value.value)
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    if name == "None":
+        return
+
+    assert scantest.arti_type_name.text() == truncate_text(part.__name__, 34)
+    assert scantest.arti_data_main.text() == str(round(part.stat_data, 1))
+
+
+@pytest.mark.parametrize(
+    "_",
+    [
+        pytest.param(None, id="face.scan.rule: Clearing artifact")
+    ]
+)
+def test_scan_arti_wipe(scantest, qtbot, _) -> None:
+    """
+    Test the clearing of the artifact
+
+    :return:
+    """
+
+    """
+    Set the user interface elements as intended
+    """
+    qtbot.mouseClick(scantest.arti_back_wipe, Qt.LeftButton)
+
+    """
+    Confirm if the user interface elements change accordingly
+    """
+    assert scantest.arti_type.currentText() == "None"
+    assert scantest.arti_levl.currentText() == "None"
+    assert scantest.arti_rare.currentText() == "Star 0"
+    assert scantest.arti_type_name.text() == "None"
+    for item in ["main", "a", "b", "c", "d"]:
+        assert getattr(scantest, f"arti_name_{item}").currentText() == "None"
+        assert getattr(scantest, f"arti_data_{item}").text() == ""
+
+
+@pytest.mark.parametrize(
+    "_",
+    [
+        pytest.param(None, id="face.scan.rule: Test OCR functionality")
+    ]
+)
+def test_scan_arti_load(scantest, qtbot, mocker, _) -> None:
+    """
+    Test OCR functionality
+
+    :return:
+    """
+
+    """
+    Create the tesseract training data
+    """
+    mocker.patch("gi_loadouts.conf.data_prefix", f"{uuid4().hex.upper()[0:8]}-")
+    make_temp_file()
+
+    """
+    Perform the action of loading the artifact information
+    """
+    temp_prm = ""
+    tempfile = "test/static/img/gi-loadouts-ocr-test.webp"
+    savefile = str(Path(gettempdir()) / "gi-loadouts-ocr-test.webp")
+    with open(tempfile, "rb") as src_file:
+        with open(savefile, "wb") as dest_file:
+            dest_file.write(src_file.read())
```

Comment:
This can lead to security vulnerabilities as it would most likely lead to a definitive path (for instance, `/tmp/gi-loadouts-ocr-test.webp` in POSIX systems). Please consider using `NamedTemporaryFile` instead of using a statically defined name.